### ISSUE TITLE
Update to Airflow >=2.0

### ISSUE THREAD

**Issue by gplssm** [openego/eGon-data#79]
Airflow 2.0 is out and brings major changes. See the [release announcement](https://github.com/apache/airflow/releases/tag/2.0.0). New important features and changes are explained in [a series of videos](https://www.youtube.com/playlist?list=PLCi-q9vYo4x-PESoBcXN0tXCMgzh5c_Pj).

We should decide whether we consider to use Airflow >=2.0 right now or if we want to wait until [2.0.1](https://github.com/apache/airflow/milestone/23) or [2.1.0](https://github.com/apache/airflow/milestone/8). In the latter cases, we should add a upper version limit for apache-airflow in the requirements list.

**Comment by gnn** [comment]
Looking at the announcement, I think we should move to 2.0. I'll try out whether any changes break our code and if they do, I'll put an upper version limit on the dependency. I'll also open a branch which migrates the existing code to 2.0, and once this is merged, we can put a lower version limit on the dependency.

**Comment by gplssm** [comment]
Once airflow is upgraded, relax the upper version limit of SQLAlchemy in setup.py. See https://github.com/openego/eGon-data/commit/13319882e695e99ca85de41290683bc5ae9dde93 for explanation.

**Comment by ClaraBuettner** [comment]
Within powerd-data we updated to airflow > 2.0 (see https://github.com/openego/powerd-data/issues/36). 
As soon as we merge the fork back to egon-data, airflow is updated.

**Comment by ClaraBuettner** [comment]
Done in powerd-data which was merged into `dev`
